src/System.Private.CoreLib/shared/System/MemoryExtensions.Split.cs

```diff
@@ -0,0 +1,105 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+namespace System
+{
+    public static partial class MemoryExtensions
+    {
+        public static SpanSplitEnumerator<T> Split<T>(this ReadOnlySpan<T> span, T separator, StringSplitOptions options = StringSplitOptions.None)
```

Comment:
StringSplitOptions is kinda weird.

Reply:
Agree. That question came up in the design review, but it wasn't resolved.
https://github.com/dotnet/corefx/issues/26528#issuecomment-420629265

Reply:
The approved API does not have `StringSplitOptions`. It is in a comment here: https://github.com/dotnet/corefx/issues/26528#issuecomment-523152939

```c#
public static SpanSplitEnumerator<char> Split(this ReadOnlySpan<char> span);
public static SpanSplitEnumerator<char> Split(this ReadOnlySpan<char> span, char separator);
public static SpanSplitEnumerator<char> Split(this ReadOnlySpan<char> span, string separator);
```

Reply:
Ah, I was confused, thanks.
What does the first overload do - it has no explicit `separator`?

Reply:
Splits on whitespace. I would follow the `string.Split(null)` rules and use `char.IsWhiteSpace`.